code/modules/assembly/flash.dm

```diff
@@ -322,6 +323,56 @@
 /obj/item/assembly/flash/cyborg/screwdriver_act(mob/living/user, obj/item/I)
 	return
 
+
+///Camera flash
+
+/obj/item/assembly/flash/camera
+	name = "camera"
+	icon = 'icons/obj/items_and_weapons.dmi'
+	desc = "A polaroid camera."
+	icon_state = "camera"
+	inhand_icon_state = "camera"
+	worn_icon_state = "camera"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	flags_1 = CONDUCT_1
+	slot_flags = ITEM_SLOT_NECK
+	pickup_sound = null
+	flash_sound = 'sound/items/polaroid1.ogg'
+	var/max_charges = 5
+	var/current_charges = 5
+	var/list/charge_timers = list()
+	var/charge_time = 300 //30 seconds
+
+/obj/item/assembly/flash/camera/burn_out()
+	return //use self charging system instead
+
+/obj/item/assembly/flash/camera/attack(mob/living/M, mob/user)
+	if(current_charges)
+		current_charges --
+		to_chat(user, span_notice("You use [src]. It now has [current_charges] charge[current_charges == 1 ? "" : "s"] remaining."))
+		charge_timers.Add(addtimer(CALLBACK(src, .proc/recharge), charge_time, TIMER_STOPPABLE))
+		..()
+	else
+		to_chat(user, span_warning("[src] is recharging."))
+	return
+
+/obj/item/assembly/flash/camera/proc/recharge(mob/user)
+	current_charges = min(current_charges+1, max_charges)
+	charge_timers.Remove(charge_timers[1])
+
+/obj/item/assembly/flash/camera/examine(mob/user)
+	. = ..()
+	. += span_notice("It has [current_charges] charges remaining.")
+	if (length(charge_timers))
+		. += "[span_notice("<b>A small display on the screen reads:")]</b>"
+	for (var/i in 1 to length(charge_timers))
+		var/timeleft = timeleft(charge_timers[i])
+		var/loadingbar = num2loadingbar(timeleft/charge_time)
+		. += span_notice("<b>CHARGE #[i]: [loadingbar] ([timeleft*0.1]s)</b>")
```

Comment:
Since it exists as a span class, may as well use it here:
```suggestion
		. += span_boldnotice("A small display on the screen reads:")
	for (var/i in 1 to length(charge_timers))
		var/timeleft = timeleft(charge_timers[i])
		var/loadingbar = num2loadingbar(timeleft/charge_time)
		. += span_boldnotice("CHARGE #[i]: [loadingbar] ([timeleft*0.1]s)")
```